### ISSUE TITLE
Starting Coroutine OnComplete Gives Index Out Of Bounds

### ISSUE THREAD

**Issue by AnthonyUccello** [Demigiant/dotween#197]
Good day!

Version: DoTween 1.1.6 & DoTweenPro 0.9.685

I am really enjoying how flexible DOTween is but I have hit a bit of a snag. In my OnComplete callback I am trying to call a courtine so I can pause for a couple seconds before moving on, here is the code:
```
obj.transform.DOScale(1, animationTime).OnComplete(
			() =>
			{
				completeCallback();
			});

// The monobehavoiur class getting callback

	public void AfterAnimationComplete()
	{
		StartCoroutine("Delay");
	}

	IEnumerator Delay()
	{
		yield return new WaitForSeconds(1f);
	}
```

If I dont start the courtine, everything works fine, the callback fires. If I do start the coroutine I get the error message with safety on:
> [Warning] DOTWEEN :: An error inside a tween callback was silently taken care of > 
>   at (wrapper managed-to-native) UnityEngine.MonoBehaviour:StartCoroutine (string,object)
>   at .AfterAnimationComplete () [0x00008] in J:\Class.cs:23 
>   at V_Main+<Class>c__AnonStorey0.<>m__0 () [0x0000d] in J:\V_Main.cs:211 
>   at DG.Tweening.Tween.OnTween

And with safety mode off I get this error:

> [Exception] NullReferenceException
> [Exception] IndexOutOfRangeException: Array index is out of range.
> [Exception] IndexOutOfRangeException: Array index is out of range.
> (wrapper stelemref) object:stelemref(object,intptr,object)

Do you have any advice on how to add a delay oncomplete (ideally without starting another animation) or an idea of what the issue is here?

Thank you!

**Comment by Demigiant** [comment]
Hi Anthony,

There should be no problem at all in running coroutines from an OnComplete callback, I do it too. The error you're seeing is because your coroutine is generating that error. Can you try to fire it outside of DOTween and see what happens?

On a secondary note, consider that if you want to just add a delay before another tween, you can either chain both tweens in a Sequence, or add a SetDelay to the second tween :)

Cheers,
Daniele

**Comment by AnthonyUccello** [comment]
@Demigiant Thanks for the fast reply!

So after some digging, it seems I was thrown off the trail. So even if I tried starting a coroutine without DOTween Unity is throwing a null error when I start a coroutine from the class.

It turns out that the class I am using to call the coroutine was a monobehaviour class I created using "new" -- which is a big nono (generally speaking). It hasn't been a problem until I tried dispatching a coroutine which apparently is breaking all threading.

So, circling back, it seems that when I fail to start a courtine because of a monobehaviour class created with the "new" keyword it throws this error. However, the reason this was not immediately apparent until your suggestion is that if I run this with DoTween and Safemode On, it doesn't throw any error at all, just a warning. It's only when I turn safe mode off that I got the error message --- and only until I tried the coroutine as you suggested without dotween I got the null right at the source 💃.

I hope this clears up the issue, thanks for the suggestion, it lead me to the source of the problem.
